packages/ui-patterns/src/CommandMenu/api/CommandInput.tsx

```diff
@@ -50,8 +52,68 @@ const CommandInput = forwardRef<
   const [inputValue, setInputValue] = useState(query)
   useEffect(() => {
     setInputValue(query)
+    previousValueRef.current = query
   }, [query])
 
+  // Get telemetry context
+  const telemetryContext = useCommandMenuTelemetryContext()
+
+  // Debounced telemetry tracking
+  const debounceTimeoutRef = useRef<NodeJS.Timeout>()
+  const previousValueRef = useRef<string>(inputValue)
+
+  const logTelemetryEvent = useCallback(
+    (value: string) => {
+      if (telemetryContext?.onTelemetry) {
+        const event = {
+          action: 'command_menu_input_typed' as const,
+          properties: {
+            value: value,
+            app: telemetryContext.app,
+          },
+          groups: {},
+        }
+        telemetryContext.onTelemetry(event)
+      }
+    },
+    [telemetryContext]
+  )
+
+  const debouncedLogTelemetry = useCallback(
+    (value: string) => {
+      // Only trigger telemetry if the user is adding characters (not removing with backspace)
+      const isAddingCharacters = value.length > previousValueRef.current.length
+
+      if (!isAddingCharacters) {
+        previousValueRef.current = value
+        return
+      }
+
+      if (debounceTimeoutRef.current) {
+        clearTimeout(debounceTimeoutRef.current)
+      }
+
+      debounceTimeoutRef.current = setTimeout(() => {
+        logTelemetryEvent(value)
+        previousValueRef.current = value
+      }, 500)
```

Comment:
```suggestion
      }, 2500)
```

We can make this much longer, since it's not providing user feedback, and there's not much value in sending all the half-typed queries

Reply:
+1